### PR TITLE
👷Clean-Up Maven Build, Adopt Wrapper, GitHub Actions

### PR DESCRIPTION
**Pull request by gilday**
There are three related changes in this PR

* **Introduce Maven Wrapper**
  Just as we're using the Gradle Wrapper in our other projects, we'll use the Maven Wrapper in this project
* **Transition from Travis-CI to GitHub Actions**
  travis-ci.org is sunsetting, and we've already partially migrated to GitHub Actions which is our CI of choice. This finished the migration
* **Clean-Up POM**
  * Use secure links
  * Manage plugin versions with `pluginManagement` (and update plugins)
  * Fix compiler targets

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -121,7 +121,6 @@
             </plugin>
             <plugin>
                 <artifactId>maven-source-plugin</artifactId>
-                <version>3.0.0</version>
```
Comment (stiemannkj1) [inline pom.xml:124]: I think we’re missing a version for the `maven-source-plugin` in `pluginManagement`.

File: pom.xml [hunk 2]
```diff
@@ -151,18 +149,16 @@
             </plugin>
             <plugin>
                 <artifactId>maven-release-plugin</artifactId>
-                <version>2.5.3</version>
                 <dependencies>
                     <dependency>
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-scm-plugin</artifactId>
-                        <version>1.8.1</version>
+                        <version>1.11.2</version>
```
Comment (stiemannkj1) [inline pom.xml]: Add property for `${maven.scm.plugin.version}` since this version is used in 2 places.

File: pom.xml [hunk 3]
```diff
@@ -108,7 +109,6 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
```
Comment (stiemannkj1) [inline pom.xml]: Remove default plugin groupId to be consistent.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+          key: ${{ runner.os }}-maven-wrapper-${{ hashFiles('./.mvn/wrapper/maven-wrapper.properties') }}
+          restore-keys: ${{ runner.os }}-maven-wrapper
+      - name: Cache Maven Packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Maven Verify
+        run: ./mvnw --batch-mode verify
```
Comment (stiemannkj1) [inline .github/workflows/build.yml]: Extremely minor: add newline to the end of `build.yml`.

File: pom.xml [hunk 5]
```diff
@@ -108,7 +109,6 @@
...
                 <artifactId>maven-gpg-plugin</artifactId>
-                <version>1.6</version>
```
Comment (stiemannkj1) [inline pom.xml:111]: Missing version for `maven-gpg-plugin` in `pluginManagement`.

**Review by stiemannkj1 — CHANGES_REQUESTED** [review]
Aside from a few missing versions, everything looks good to me.

Looks like there’s an old `publish.yml` Github Actions file in the repo as well. Should that be removed?

**Review by BrianPhillips2020 — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (gilday) [reply pom.xml]: I don't know why we're including the `maven-scm-plugin` dependency here. The `maven-release-plugin` does not depend on the `maven-scm-plugin`. I'm thinking we take it out and deal with the consequences later.

File: .github/workflows/build.yml (hunk 4 above)
Comment (gilday) [reply .github/workflows/build.yml]: The formatting change I'll make next would have fixed this!

**Comment by gilday** [comment]
> Looks like there’s an old `publish.yml` Github Actions file in the repo as well. Should that be removed?

No, that is how we release this project to Maven Central. I'm working on fixing the release process 🔜

**Review by stiemannkj1 — APPROVED** [review]

**Pull request merged**